format-common/src/main/java/io/cdap/plugin/format/input/AbstractEmptyInputFormat.java

```diff
@@ -30,7 +30,7 @@
  * @param <K> the key class
  * @param <V> the value class
  */
-public class EmptyInputFormat<K, V> extends InputFormat<K, V> {
+public class AbstractEmptyInputFormat<K, V> extends InputFormat<K, V> {
```

Comment:
Lets not put a normal class name as abstract class.

Reply:
Made the class abstract !

Reply:
But then you have to pass a concrete class in getEmptyInputFormatClassName. Do that

Reply:
Can we make `getEmptyInputFormatClassName` abstract then ?

Reply:
no, then we have to give implementation for all plugins. Have a blank Input format class for now and return that, in future once all the plugins have their own classes we can make it abstract.

Reply:
Ok , added a new class `EmptyInputFormat`.

Ideally we should have made this abstract right ?
It feels wired to have a concrete class being used inside an abstract class, when the impl is not internal.

Reply:
We should make it abstract in case every child class have implementation for it. Here we don't have that liberty and existing plugins will falter.